integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/container/DorisContainer.java

```diff
@@ -0,0 +1,151 @@
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.rnorth.ducttape.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.Network;
+
+public class DorisContainer extends BaseContainer {
+  public static final Logger LOG = LoggerFactory.getLogger(DorisContainer.class);
+
+  public static final String DEFAULT_IMAGE = System.getenv("GRAVITINO_CI_DORIS_DOCKER_IMAGE");
+  public static final String HOST_NAME = "gravitino-ci-doris";
+  public static final String USER_NAME = "root";
+  public static final String PASSWORD = "root";
+  public static final int FE_HTTP_PORT = 8030;
+  public static final int FE_MYSQL_PORT = 9030;
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  protected DorisContainer(
+      String image,
+      String hostName,
+      Set<Integer> ports,
+      Map<String, String> extraHosts,
+      Map<String, String> filesToMount,
+      Map<String, String> envVars,
+      Optional<Network> network) {
+    super(image, hostName, ports, extraHosts, filesToMount, envVars, network);
+  }
+
+  @Override
+  protected void setupContainer() {
+    super.setupContainer();
+    withLogConsumer(new PrintingContainerLog(format("%-14s| ", "DorisContainer")));
+    withStartupTimeout(Duration.ofMinutes(10));
+  }
+
+  @Override
+  public void start() {
+    super.start();
+    Preconditions.check("Doris container startup failed!", checkContainerStatus(5));
+    Preconditions.check("Doris container password change failed!", changePassword());
+  }
+
+  @Override
+  protected boolean checkContainerStatus(int retryLimit) {
+    int nRetry = 0;
+    boolean isDorisContainerReady = false;
+
+    String dorisMySQLUrl = format("jdbc:mysql://%s:%d/", getContainerIpAddress(), FE_MYSQL_PORT);
+    LOG.info("Doris url is " + dorisMySQLUrl);
+    Connection connection = null;
+
+    while (nRetry++ < retryLimit) {
+      try {
+        connection = DriverManager.getConnection(dorisMySQLUrl, USER_NAME, "");
+
+        // execute `SHOW PROC '/backends';` to check if backends is ready
+        String query = "SHOW PROC '/backends';";
+        Statement statement = connection.createStatement();
+        ResultSet resultSet = statement.executeQuery(query);
+        while (resultSet.next()) {
+          String alive = resultSet.getString("Alive");
+          if (alive.equalsIgnoreCase("true")) {
+            LOG.info("Doris container startup success!");
+            isDorisContainerReady = true;
+            break;
+          }
+        }
+
+        if (isDorisContainerReady) {
+          break;
+        }
+        LOG.info("Doris container is not ready yet!");
+        Thread.sleep(5000);
+      } catch (Exception e) {
+        LOG.error(e.getMessage(), e);
+      }
+    }
+
+    if (connection != null) {
+      try {
+        connection.close();
+      } catch (SQLException e) {
+        LOG.warn(e.getMessage(), e);
+      }
+    }
```

Comment:
You should use try-with-resource to ensure that the connection is properly closed after use. For example:
```java
protected boolean checkContainerStatus(int retryLimit) {
    String dorisMySQLUrl = format("jdbc:mysql://%s:%d/", getContainerIpAddress(), FE_MYSQL_PORT);
    LOG.info("Doris url is " + dorisMySQLUrl);

    for (int nRetry = 0; nRetry < retryLimit; nRetry++) {
        try (Connection connection = DriverManager.getConnection(dorisMySQLUrl, USER_NAME, "")) {
             Statement statement = connection.createStatement()) {

            String query = "SHOW PROC '/backends';";
            ResultSet resultSet = statement.executeQuery(query);
            while (resultSet.next()) {
                String alive = resultSet.getString("Alive");
                if ("true".equalsIgnoreCase(alive)) {
                    LOG.info("Doris container startup success!");
                    return true;
                }
            }

            LOG.info("Doris container is not ready yet!");
            Thread.sleep(5000);
        } catch (SQLException | InterruptedException e) {
            LOG.error(e.getMessage(), e);
            // ...
        }
    }

    return false;
}
```